src/tmpfiles/tmpfiles.c

```diff
@@ -794,6 +814,19 @@ static int dir_cleanup(
                                            cutoff_nsec, sub_path, age_by_file, false))
                                 continue;
 
+                        fd = xopenat_nomod(dirfd(d), de->d_name);
+                        if (fd < 0) {
+                                if (fd != -ENOENT)
+                                        r = log_warning_errno(fd, "Opening file \"%s\" failed, ignoring: %m", sub_path);
+
+                                continue;
+                        }
+
+                        if (flock(fd, LOCK_EX|LOCK_NB) < 0) {
+                                log_debug_errno(errno, "Couldn't acquire shared BSD lock on file \"%s\", skipping: %m", p);
```

Comment:
hmm, I think we should skip the file only if we get EAGAIN, and if it fails otherwise ignore the error. there might be file systems that don't implement locking (think: NFS client against an NFS server that doesn't support it). I think in such a case it is better to still clean up the files instead of failing altogether.